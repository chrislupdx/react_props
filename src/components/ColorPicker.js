import React from 'react';
import propTypes from 'prop-types';
import styles from './ColorPicker.css';

function ColorPicker({ selectColorHandler }) {
  return (
    <section className={styles.colorPicker}>
      <button className={styles.red} onClick={() => selectColorHandler('red')}>Red</button>
      <button className={styles.yellow} onClick={() => selectColorHandler('yellow')}>Yellow</button>
      <button className={styles.blue} onClick={selectColorHandler.bind(null, 'blue')}>Blue</button>
    </section>
  );
}

ColorPicker.propTypes = {
  selectColorHandler: propTypes.func.isRequired
};

export default ColorPicker;
