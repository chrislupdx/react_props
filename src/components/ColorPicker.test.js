import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('colorpicker component', () => {
  it('renders a colorpicker', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles the red button press', () => {
    // intercept console.log calls
    console.log = jest.fn();
    const wrapper = shallow(<ColorPicker />);

    wrapper.find('button').at(0).simulate('click');

    expect(console.log).toHaveBeenCalledWith('red');
  });
  
});

