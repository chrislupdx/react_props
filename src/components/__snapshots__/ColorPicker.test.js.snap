// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders ColorPicker 1`] = `
<section
  className="colorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    Red
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    Yellow
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue
  </button>
</section>
`;
